### ISSUE TITLE
Preconditionning

### ISSUE THREAD

**Issue by rveltz** [Jutho/KrylovKit.jl#7]
Hi,

I have become addicted to your package. The downside is that I would like an additional feature ;)
I use Arnoldi iterations for  computing the spectrum and the convergence is super slow.
So, I was wondering how difficult it would be for you to add support for preconditionning?

Thank you a lot for your help,

Best regards

**Comment by Jutho** [comment]
Thanks. I think the hardest part is coming up with a nice and convenient interface. Unfortunately, I don't have much time for this right now. I would certainly be welcome to any contributions.

**Comment by Jutho** [comment]
No need to close this; it's good to let feature requests hang around as open issues.

**Comment by rveltz** [comment]
Don't you like the interface from `IterativeSolvers`? ie a keywork argument `Pl = Idendity` where one could use functions or matrices. Is it a lot of work to add? Especially to `eigsolve`?

**Comment by Jutho** [comment]
It's rather an issue of a lack of expertise from my side. I have no experience with preconditioners, as there are typically no obvious choices in the cases where I am solving eigenvalue and linear problems. And if there are, they are usually incorporated in a more explicit way, i.e. by transforming to a new set of variables. 

The lack of self-need is also why I can't afford to invest too much time in this, at least not right now. But I do welcome any contribution in that direction.

**Comment by Jutho** [comment]
Exactly, in the applications I am working with, we often have a generalized eigenvalue problem that, by parameterizing the problem in terms of a more suitable set of variables, is transformed into a normal eigenvalue problem.
